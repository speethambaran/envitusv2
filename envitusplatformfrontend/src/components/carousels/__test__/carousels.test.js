import React from 'react';
import { shallow } from 'enzyme';
import Carousels from './../carousels';
import { findByDataTest } from './../../../Utils';

const preConfig = (props={}) => {
    const component = shallow(<Carousels {...props} />);
    return component;
};

describe('Carousels Test', () => {
    it('Redering Properly with prop values', () => {
        let component;
        const props = {
            items: [{
                id: "testBuilding",
                img: "dummypath",
                alt: "testAltBuilding"
            }] 
        };
        component = preConfig(props);
        const wrapperCarousal = findByDataTest(component, 'carouselComponent');
        expect(wrapperCarousal.length).toBe(1);
        const wrapperCarousalItem = findByDataTest(component, 'carouselItemComponent');
        expect(wrapperCarousalItem.length).toBe(1);
        const wrapperImg = findByDataTest(component, 'ImgComponent');
        expect(wrapperImg.length).toBe(1);
        expect(wrapperCarousal).toMatchSnapshot();
    });

});