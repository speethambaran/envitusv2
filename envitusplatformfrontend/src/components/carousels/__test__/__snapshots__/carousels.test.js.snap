// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousels Test Redering Properly with prop values 1`] = `
<Bootstrap(undefined)
  data-test="carouselComponent"
>
  <CarouselItem
    data-test="carouselItemComponent"
    key="testBuilding"
  >
    <img
      alt="testAltBuilding"
      className="d-block mx-auto"
      data-test="ImgComponent"
      src="dummypath"
    />
  </CarouselItem>
</Bootstrap(undefined)>
`;
